feat: handle foreground push notifications (#161)
TheWidlarzGroup/react-native-notificated: docs/docs/intro/examples/push-notifications-example.md

---
sidebar_position: 7
---

# 📭 Push notifications example

<br/>

>     We prepared examples of how to implement push notifications in Your app, using <a href="https://rnfirebase.io/">raect native firebase</a>.<br/>

Reviewer: ```suggestion
Here's an example of how you can handle push notifications in your application (when in foreground) using [Firebase](https://rnfirebase.io/)
```

    Let's go then! 💪 
<br/>

## 💡 Setup

<br/>
    Before we jump into setup proccess, make sure that you have created a new firebase project. You will need it later to send push notifications. If you haven't done it yet, you can do it <a href="https://console.firebase.google.com/">here</a>. 
<br/>
<br/>

>     Ok, after we have created a new firebase project, it's time to proceed to installing two packages that will be necessary to handle push notifications:

Reviewer: ```suggestion
With a new firebase project created, it's time to proceed with installation of these two packages that will be necessary to handle push notifications:
```

    <br/>- <a href="https://rnfirebase.io/#prerequisites">@react-native-firebase/app</a>
    <br/>- <a href="https://rnfirebase.io/messaging/usage#installation">@react-native-firebase/messaging</a>
    <br/>
    <br/> 
<br/>

> 

Reviewer: - `<br />` -> just a double `ENTER`
- `<a href=></a>` -> `[React Native Firebase](https://rnfirebase.io/)`

https://www.markdownguide.org/basic-syntax/

:::info
Documentation clearly describes the whole process of installation, so I will not focus on it here.
Before proceeding to the next step, make sure you have done all previous steps.

> 

Reviewer: ```suggestion
```

:::

## 👀 Example Implementation

### App in Foreground

> To handle push notifications in foreground we have to create a listener which will listen for incoming calls from firebase.

Reviewer: ```suggestion
To handle push notifications in foreground we have to create a listener which will subscribe for all incoming firebase push events.
```

We can do this by using `messaging().onMessage()` from `@react-native-firebase/messaging`. Method `onMessage()`

> takes asynchronous functions as a parameter and pass to it, as a result of promise, object with push notification message.

Reviewer: ```suggestion
takes an asynchronous function as a parameter, with an argument which is the push notification payload.
```

> Now we can use this result object to set our notification with `notify()`. Below You will find basic example how to handle it:

Reviewer: ```suggestion
Now we can use this data to set our in-app notification with a `notify()`:
```
